service/service.go

```diff
@@ -172,6 +172,13 @@ func (srv *Service) Shutdown(ctx context.Context) error {
 	return errs
 }
 
+func (srv *Service) ConfigResolved(_ context.Context, effectiveConfig map[string]interface{}) error {
```

Comment:
I decided to make a new method on the service for notifying extensions of changes in the configuration since we would also need a method to notify the service of more granular changes to the configuration if the Collector gains that capability. If we only expect the Collector to do full reloads of the service, then we could likely remove this method.

Reply:
You cannot remove something but you can always add. So please don't add anything until is necessary.